apps/api/src/app/auth/services/passport/jwt.strategy.ts

```diff
@@ -29,17 +29,33 @@ export class JwtStrategy extends PassportStrategy(Strategy) {
       throw new UnauthorizedException();
     }
 
-    await this.resolveEnvironmentId(req, session);
+    const environmentId = this.resolveEnvironmentId(req, session);
+
+    // eslint-disable-next-line no-param-reassign
+    session.environmentId = environmentId;
+
+    if (session.environmentId) {
+      const environment = await this.environmentRepository.findOne(
+        {
+          _id: session.environmentId,
+          _organizationId: session.organizationId,
+        },
+        '_id'
+      );
+
+      if (!environment) {
+        throw new UnauthorizedException('Cannot find environment');
```

Comment:
Let's pass the session here to ease debugging from Sentry.

```suggestion
        throw new UnauthorizedException('Cannot find environment', { data: session })
```